x/emissions/keeper/inference_synthesis/network_inferences.go

```diff
@@ -71,55 +69,131 @@ func GetNetworkInferencesAtBlock(
-			return networkInferences, nil, infererWeights, forecasterWeights, nil
+			Logger(ctx).Warn(fmt.Sprintf("Error constructing network inferences builder topic: %s", err.Error()))
+			return nil, nil, infererWeights, forecasterWeights, err
+		}
+		networkInferences = networkInferenceBuilder.CalcAndSetNetworkInferences().Build()
+
+		forecastImpliedInferencesByWorker = networkInferenceBuilder.palette.ForecastImpliedInferenceByWorker
+		infererWeights = networkInferenceBuilder.weights.inferers
+		forecasterWeights = networkInferenceBuilder.weights.forecasters
+	} else {
+		// If there is only one valid inference, then the network inference is the same as the single inference
+		// For the forecasts to be meaningful, there should be at least 2 inferences
+		singleInference := inferences.Inferences[0]
+
+		networkInferences = &emissions.ValueBundle{
+			TopicId:       topicId,
+			CombinedValue: singleInference.Value,
+			InfererValues: []*emissions.WorkerAttributedValue{
+				{
+					Worker: singleInference.Inferer,
+					Value:  singleInference.Value,
+				},
+			},
+			ForecasterValues:       []*emissions.WorkerAttributedValue{},
+			NaiveValue:             singleInference.Value,
+			OneOutInfererValues:    []*emissions.WithheldWorkerAttributedValue{},
+			OneOutForecasterValues: []*emissions.WithheldWorkerAttributedValue{},
+			OneInForecasterValues:  []*emissions.WorkerAttributedValue{},
+		}
+	}
+
+	return networkInferences, forecastImpliedInferencesByWorker, infererWeights, forecasterWeights, nil
+}
+
+func GetLatestNetworkInference(
+	ctx sdk.Context,
+	k keeper.Keeper,
+	topicId TopicId,
+) (
+	*emissions.ValueBundle,
+	map[string]*emissions.Inference,
+	map[string]alloraMath.Dec,
+	map[string]alloraMath.Dec,
+	error,
+) {
+	networkInferences := &emissions.ValueBundle{
+		TopicId:          topicId,
+		InfererValues:    make([]*emissions.WorkerAttributedValue, 0),
+		ForecasterValues: make([]*emissions.WorkerAttributedValue, 0),
+	}
+	forecastImpliedInferencesByWorker := make(map[string]*emissions.Inference, 0)
+	var infererWeights map[string]alloraMath.Dec
+	var forecasterWeights map[string]alloraMath.Dec
+
+	inferences, infBlockHeight, err := k.GetLatestTopicInferences(ctx, topicId)
+	if err != nil {
+		fmt.Printf("Error getting inferences: %s", err.Error())
+		return networkInferences, nil, infererWeights, forecasterWeights, nil
+	}
+	forecasts, err := k.GetForecastsAtBlock(ctx, topicId, infBlockHeight)
+	if err != nil {
+		fmt.Printf("Error getting forecasts: %s", err.Error())
+		return networkInferences, nil, infererWeights, forecasterWeights, nil
+	}
+	if len(inferences.Inferences) > 1 {
+		moduleParams, err := k.GetParams(ctx)
+		if err != nil {
+			return nil, nil, infererWeights, forecasterWeights, err
 		}
 		topic, err := k.GetTopic(ctx, topicId)
 		if err != nil {
-			Logger(ctx).Warn(fmt.Sprintf("Error getting topic: %s", err.Error()))
+			fmt.Printf("Error getting topic: %s", err.Error())
```

Comment:
it looks like we are doing loger.warn in some places and fmt.printf in others? Let's just pick on and stick to it (I don't have a strong opinion, but I think logger.warn is what we want right?)